llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -1198,6 +1198,18 @@ bool CombinerHelper::matchCombineExtractedVectorLoad(MachineInstr &MI,
   if (!VecEltTy.isByteSized())
     return false;
 
+  // Check for load fold barriers between the extraction and the load.
+  if (MI.getParent() != LoadMI->getParent())
```

Comment:
This check should not be necessary. `MI` and `LoadMI` are in the same function.

Reply:
They are in the same function, but not necessarily in the same block, right?

Reply:
Sorry. I misunderstood `Parent`. You are right, they must be in the same block.